dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/JdbcEventStore.java

```diff
@@ -603,141 +600,113 @@ lateral jsonb_each(
   }
 
   /**
-   * Generates a single INNER JOIN for each attribute we are filtering or ordering on. We can search
-   * by a range of operators. All searching is using lower() since attribute values are
-   * case-insensitive.
+   * Generates the WHERE-clause related to the user provided attribute filters. It will find the
+   * tracked entity attributes that match the given user filter criteria. This condition only
+   * applies when an attribute filter is specified.
    */
-  private String joinAttributeValue(
-      EventQueryParams params, MapSqlParameterSource mapSqlParameterSource) {
-    StringBuilder sql = new StringBuilder();
-
+  private String getWhereClauseFromAttributeFilterConditions(
+      EventQueryParams params, MapSqlParameterSource mapSqlParameterSource, SqlHelper hlp) {
+    StringBuilder fromBuilder = new StringBuilder();
     for (Entry<TrackedEntityAttribute, List<QueryFilter>> queryItem :
         params.getAttributes().entrySet()) {
-
       TrackedEntityAttribute tea = queryItem.getKey();
       String teaUid = tea.getUid();
       String teaValueCol = quote(teaUid);
-      String teaCol = quote(teaUid + "ATT");
 
-      sql.append(" inner join trackedentityattributevalue ")
-          .append(teaValueCol)
-          .append(" on ")
-          .append(teaValueCol + ".trackedentityid")
-          .append(" = TE.trackedentityid ")
-          .append(" inner join trackedentityattribute ")
-          .append(teaCol)
-          .append(" on ")
-          .append(teaValueCol + ".trackedentityattributeid")
-          .append(EQUALS)
-          .append(teaCol + ".trackedentityattributeid")
+      fromBuilder
+          .append(hlp.whereAnd())
+          .append(" TE.trackedentityid is not null ") // filters out results from event programs
           .append(AND)
-          .append(teaCol + ".UID")
-          .append(EQUALS)
-          .append(singleQuote(teaUid));
-
-      sql.append(
-          getAttributeFilterQuery(
-              mapSqlParameterSource,
-              queryItem.getValue(),
-              teaCol,
-              teaValueCol,
-              tea.getValueType().isNumeric()));
+          .append(SPACE)
+          .append(
+              getAttributeFilterQuery(
+                  mapSqlParameterSource,
+                  queryItem.getValue(),
+                  teaValueCol,
+                  tea.getValueType().isNumeric()));
     }
-
-    return sql.toString();
+    return fromBuilder.toString();
   }
 
   private String getAttributeFilterQuery(
       MapSqlParameterSource mapSqlParameterSource,
       List<QueryFilter> filters,
-      String teaCol,
       String teaValueCol,
       boolean isNumericTea) {
 
     if (filters.isEmpty()) {
       return "";
     }
 
-    StringBuilder query = new StringBuilder(AND);
-    // In SQL the order of expressions linked by AND is not
-    // guaranteed.
-    // So when casting to number we need to be sure that the value
-    // to cast is really a number.
```

Comment:
Since you removed the comment I assume this not apply anymore. Why? Please explain that change in the SQL. Seems like you removed the `case`.

Reply:
Someone correct me if I'm missing something, but we are already checking if the TEA type is numeric `[here](https://github.com/dhis2/dhis2-core/pull/20128/files#diff-68b5ae1d0a8364a34a53140313e32a1ecc8eb172c131a37abe2d136fe1244191R625)`

This flag is then used [here](https://github.com/dhis2/dhis2-core/pull/20128/files#diff-68b5ae1d0a8364a34a53140313e32a1ecc8eb172c131a37abe2d136fe1244191R646) and [here](https://github.com/dhis2/dhis2-core/pull/20128/files#diff-68b5ae1d0a8364a34a53140313e32a1ecc8eb172c131a37abe2d136fe1244191R659) to build the query.

That's why I don't think this `case` is needed anymore.

Reply:
and there is a test for this?

Reply:
yes, we have tests that cover the TEA being numeric and text